### ISSUE TITLE
UnderStand Mask model to _get_action_masks in LogitsProcessor 

### ISSUE THREAD

**Issue by xesdiny** [allenai/RL4LMs#31]
In this line[code](https://github.com/allenai/RL4LMs/blob/aa5d337c4c587049e039d572042bf5c95926c3be/rl4lms/envs/text_generation/hf_generation_utils.py#L1980)
I saw that the code sets `MaskLogitsProcessorCasualLM` Init process uses `deepcopy(self._policy_model).eval()` and during the generate process, `GenerationMixinWithRawScores.sampler` executes pre-process distribution and Hook calls to the custom `LogitsProcessor`. I compared `next_token_logits_raw` from policy_model is indeed different from `next_token_logits` from mask_model in the same generate pipeline, what is the meaning of doing here? I really want to know?
